Running-Mastodon/Production-guide.md

```diff
@@ -1,20 +1,190 @@
-Production guide
-================
+# Mastodon Production Guide
 
-The following HTTP headers are already set internally and should not be set again:
+**Disclaimer:**
 
+This guide was written for [Ubuntu Server 16.04](https://www.ubuntu.com/server), you may run into issues if you are using another operating system. We are welcoming contributions for guides to other distributions.
+
+This document is also written with the expectation that you have a technical level high enough to administrate Linux servers.
+
+## What is this guide?
+
+This guide is a walk through of the setup process of a [Mastodon](https://github.com/tootsuite/mastodon/) instance.
+
+We use example.com to represent a domain or sub-domain. Example.com should be replaced with your instance domain or sub-domain.
+
+## Prerequisites
+
+You will need the following for this guide:
+
+- A server running [Ubuntu Server 16.04](https://www.ubuntu.com/server).
+- Root access to the server.
+- A domain or sub-domain to use for the instance.
+
+## DNS
+
+DNS records should be added before anything is done on the server.
+
+The records added are:
+
+-  A record (IPv4 address) for example.com
+-  AAAA record (IPv6 address) for example.com
+
+> ### A Helpful And Optional Note
+>
+> Using `tmux` when following through with this guide will be helpful.
+> 
+>
+> Not only will this help you not lose your place if you are disconnected, it will let you have multiple terminal windows open for switching contexts (root user versus the mastodon user).
+>
+> You can install [tmux](https://github.com/tmux/tmux/wiki) from the package manager:
+>
+> ```sh
+> apt -y install tmux
+> ```
+
+## Dependency Installation
+
+All dependencies should be installed as root.
+
+### node.js Repository
+
+You will need to add an external repository so we can have the version of [node.js](https://nodejs.org/en/) required.
+
+Download this script:
+
+```sh
+wget https://deb.nodesource.com/setup_6.x
+```
+
+Review the script download before running:
+
+```sh
+less setup_6.x
+```
+
+Once you have reviewed the script, run it
+
+```sh
+bash setup_6.x
+```
+
+The [node.js](https://nodejs.org/en/) repository is now added.
```

Comment:
Asking users to inspect the script seems a bit excessive – will they even know what to look for? The security implications of `wget | sudo sh` have been debated endlessly, but IMHO it's no more insecure than an `apt install` which is also downloading bits off the network via HTTPS and executing code on your machine as root.

I'd prefer to make things simpler for users and just put [the NodeSource instructions](https://github.com/nodesource/distributions) exactly as-is, i.e. `curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -`.